block-producer/src/test_utils/block.rs

```diff
@@ -140,15 +140,27 @@ impl MockBlockBuilder {
     }
 
     pub fn build(self) -> Block {
-        let created_notes = self.created_notes.unwrap_or_default();
+        let created_note_envelopes = self.created_note_envelopes.unwrap_or_default();
+
+        let created_notes = created_note_envelopes
+            .iter()
+            .map(|(batch_index, note_index, note_envelope)| NoteCreated {
```

Comment:
Yes I saw that one, considering it was in test_utils I did not find it crucial to DRY an implement something that would make the code slightly less readable.

What do you think?

Reply:
These 2 situations are not similar if you look into the code. 

We will need to refactor the code in the future to be able to get the same architecture everywhere.

Reply:
My idea was to implement such conversion:
```rust
impl From<(usize, usize, &NoteEnvelope, Option<Vec<u8>>)> for NoteCreated {
...
```
This conversion fill fit well for both cases. The only concern I have, that tuple with 4 fields might be too long.

Reply:
I don't think that this conversion is truly needed in this case. And I do believe that the tuple is too long to be used as is if we decide to add a `From` then I would recommend using a dedicated type to represent this tuple, but I believe that it should not be needed because the code is clear enough for now and the code that we are optimizing for right now will disapear in a near refactor.